front-end/src/app/shared/models/contact.model.ts

```diff
@@ -73,42 +73,100 @@ export class Contact extends BaseModel {
+  inContacts: boolean;
+}
+
+export class FecApiCommitteeLookupData {
   id: string | null = null;
   name: string | null = null;
+
   // prettier-ignore
-  static fromJSON(json: any): FecCommitteeLookupData { // eslint-disable-line @typescript-eslint/no-explicit-any
-    return plainToClass(FecCommitteeLookupData, json);
+  static fromJSON(json: any): FecApiCommitteeLookupData { // eslint-disable-line @typescript-eslint/no-explicit-any
+    return plainToClass(FecApiCommitteeLookupData, json);
   }
-  static toSelectItem(data: FecCommitteeLookupData): SelectItem<string> {
+
+  static toSelectItem(data: FecApiCommitteeLookupData,
+    inContacts: boolean): ContactLookupSelectItem<FecApiCommitteeLookupData> {
     return (
       data && {
         label: `${data.name} (${data.id})`,
-        value: `${data.id}`,
+        value: data,
+        inContacts: inContacts,
+      }
+    );
+  }
+}
+
+export class FecfileCommitteeLookupData extends Contact {
+  static toSelectItem(data: FecfileCommitteeLookupData,
+    inContacts: boolean): ContactLookupSelectItem<Contact> {
+    return (
+      data && {
+        label: `${data.name} (${data.committee_id})`,
+        value: data,
+        inContacts: inContacts,
       }
     );
   }
 }
 
 export class CommitteeLookupResponse {
-  fec_api_committees: FecCommitteeLookupData[] | null = null;
-  fecfile_committees: FecCommitteeLookupData[] | null = null;
+  fec_api_committees: FecApiCommitteeLookupData[] | null = null;
+  fecfile_committees: FecfileCommitteeLookupData[] | null = null;
+
   // prettier-ignore
   static fromJSON(json: any): CommitteeLookupResponse { // eslint-disable-line @typescript-eslint/no-explicit-any
     return plainToClass(CommitteeLookupResponse, json);
   }
+
   toSelectItemGroups(): SelectItemGroup[] {
-    const fecApiSelectItems = this.fec_api_committees?.map((data) => FecCommitteeLookupData.toSelectItem(data)) || [];
-    const fecfileSelectItems = this.fecfile_committees?.map((data) => FecCommitteeLookupData.toSelectItem(data)) || [];
+    const fecApiSelectItems = this.fec_api_committees?.map((data) =>
+      FecApiCommitteeLookupData.toSelectItem(data, false)) || [];
+    const fecfileSelectItems = this.fecfile_committees?.map((data) =>
+      FecfileCommitteeLookupData.toSelectItem(data, true)) || [];
     return [
       {
         label: 'Select an existing committee contact:',
         items: fecfileSelectItems,
       },
       {
-        label: 'Create a new contact from list of registered committee:',
+        label: 'Create a new contact from list of registered committees:',
         items: fecApiSelectItems,
       },
     ];
   }
 }
+
+export class FecfileIndividualLookupData extends Contact {
+  static toSelectItem(data: FecfileIndividualLookupData,
+    inContacts: boolean): ContactLookupSelectItem<Contact> {
+    return (
+      data && {
+        label: `${data.last_name}, ${data.first_name}`,
+        value: data,
+        inContacts: inContacts,
+      }
```

Comment:
What happens when data is null or undefined? Does the toSelectItem() return the boolean false?

Reply:
I reworked the model classes to use 'this' instead and change this method from static, so it should now never be null.